packages/store/src/lib/elf-hooks.spec.ts

```diff
@@ -0,0 +1,59 @@
+import { createStore } from './create-store';
+import { elfHooks, elfHooksRegistry } from './elf-hooks';
+import { select } from './operators';
+import { withProps } from './props.state';
+
+describe('elfHooks', () => {
+  describe('METHOD: registerPreStoreUpdate', () => {
+    it('preStoreUpdate should be called if defined', () => {
+      elfHooks.registerPreStoreUpdate(() => {
+        return { filter: 'elf' };
```

Comment:
I'm not sure what you did here. This now will always be the new state, no?

Reply:
That's right. Although this is not a realistic scenario, I think it's OK for the purposes of the unit test to assert that the hook returns the expected state. line 29 of this spec assures of that.